src/net.h

```diff
@@ -943,11 +943,14 @@ class CNode
     void MaybeSetAddrName(const std::string& addrNameIn);
 };
 
-
-
-
-
 /** Return a timestamp in the future (in microseconds) for exponentially distributed events. */
 int64_t PoissonNextSend(int64_t now, int average_interval_seconds);
 
+/** Wrapper to return mockable type */
+inline std::chrono::microseconds PoissonNextSend(std::chrono::microseconds now, std::chrono::microseconds average_interval)
+{
+    int64_t interval_seconds = (std::chrono::duration_cast<std::chrono::seconds>(average_interval)).count();
+    return std::chrono::microseconds{PoissonNextSend(now.count(), interval_seconds)};
```

Comment:
in commit e768ad10fbd24929059a1ada67cefaa3bc570b0d:

I don't think casting (truncating) microseconds to seconds is acceptable, as it degrades everything smaller than a second to a delay of zero.

You could just pass in `std::chrono::seconds average_interval` and then call here:

```cpp
     return std::chrono::microseconds{PoissonNextSend(now.count(), interval_seconds.count())};